### DISCUSSION TITLE
Browser refresh -> state loss

### DISCUSSION THREAD

**Discussion by joelwiesmann** [writer/writer-framework#183] [Q&A]
Hi all

I'm creating a small multipage Streamsync application. 
0. (root) state is initalized with empty hash value, user is forwarded on login page
1. (login) The user provides a hash on the login page which is saved to the StateProxy.
2. (login) The user gets forwarded to a control-page. 
3. (control) Control-page verifies, whether the StateProxy contains the hash (ss-page-open), if not, the user gets forwarded to login page (step 1)

What I observed now is, that the user, once successfully on the control page, looses the state, if he refreshes the browser page. It seems like the user gets the initial_state assigned again. I verified this by putting a print statement in the main.py which shows up twice - once when the user initially access the application, and a second time, when the control page is refreshed.

My environment is Windows and I tried to run the application using the streamsync edit & run commands. Both seem to behave the same. 

Anyone has an idea why I'm loosing the users session or how I can debug this further?

**Comment by FabienArcellier** [comment]
I have noticed the user session is reset on each browser refresh. I haven't found a way to save the session id in a cookie to preserve it.

A ``set_cookie`` or ``set_session`` function has to be developped on the ``StreamsyncState`` object.

This would allow us to implement a mechanism that would work in this way. It's a prototype, a robust session mechanism could be build over it.

```python
session_store = {}

def page_open(state, session):
    if 'session' not in session['cookies']:
        state.set_cookie('session', uuid.uuid4().hex)
    else:
        session_id = session['cookies']['session']
        user_session = session_store[session_id]
    
        if 'image' in user_session:
            state["image"] = user_session['image']

def click_img(state, context, session):
    session_id = session['cookies']['session']
    user_session = session_store[session_id]
    
    user_session['image'] =  context['item']
    state["image"] = context['item']
```

Maybe @ramedina86 has a way to handle this sort of usecase.

**Reply by JE-VS** [reply]
Such feature would be great, indeed!

**Reply by polishdataengineer** [reply]
Will you try to implement this feature? It would be very helpful.

**Reply by polishdataengineer** [reply]
@FabienArcellier @ramedina86 Will you try to implement this great feature?

**Comment by Xynonners** [comment]
is there any way to store data on the user / browser side yet?

**Reply by FabienArcellier** [reply]
We haven't yet :(. What would be your usecase ?
